src/plugins/workspace/common/constants.ts

```diff
@@ -158,3 +160,9 @@ export enum AssociationDataSourceModalMode {
   DirectQueryConnections = 'direction-query-connections',
 }
 export const USE_CASE_PREFIX = 'use-case-';
+
+// Workspace will handle both data source and data connection type saved object.
+export const WORKSPACE_DATA_SOURCE_AND_CONNECTION_OBJECT_TYPES = [
```

Comment:
any thought about using the data structure type? 

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/common/datasets/types.ts#L111

example:

https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/query_enhancements/public/datasets/s3_type.ts#L181

Reply:
Looks good, will take a look on whether we could use this.

Reply:
1. Why isn't type in https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/common/datasets/types.ts#L41  `data-source`?
2. Is there an example for `data-connection` type object?
3. Im thinking whether data structure type would be a little complex for workspace? Because workspace only cares about saved object related attributes and doesn't care about connection details.